client/src/components/mapping/FhirMappingPanel/FhirResourceTree/nodeLabel.tsx

```diff
@@ -2,39 +2,59 @@ import React from 'react';
 import { Menu, MenuItem, ContextMenu, Tooltip } from '@blueprintjs/core';
 import { Attribute } from '@arkhn/fhir.ts';
 
+import AddExtensionSelect from 'components/selects/addExtensionSelect';
+import { ApolloProvider, useApolloClient } from 'react-apollo';
+
 interface NodeLabelProps {
   attribute: Attribute;
   addNodeCallback: any;
+  addExtensionCallback: any;
   deleteNodeCallback: any;
 }
 
 export const NodeLabel = ({
-  attribute: { types, definition, id, isArray, isItem, isSlice },
+  attribute: { types, definition, id, isArray, isItem, isSlice, extensions },
   addNodeCallback,
+  addExtensionCallback,
   deleteNodeCallback
 }: NodeLabelProps) => {
+  const client = useApolloClient();
+  const renderAddItem = () => (
+    <MenuItem icon={'add'} onClick={addNodeCallback} text={'Ajouter un item'} />
+  );
+
+  const renderAddExtension = () => (
+    <AddExtensionSelect
+      loading={false}
+      disabled={false}
+      items={extensions}
+      onChange={addExtensionCallback}
+    />
+  );
+
+  const renderRemoveItem = () => (
+    <MenuItem
+      icon={'delete'}
+      onClick={deleteNodeCallback}
+      text={"Supprimer l'item"}
+    />
+  );
+
   const showContextMenu = async (e: React.MouseEvent<HTMLDivElement>) => {
     e.preventDefault();
+    const hasExtensions = !isArray && !!extensions && extensions.length > 0;
```

Comment:
Don't you think it's confusing to call it extensions instead of something like allowedExtensions?

Reply:
changed